tests/conftest.py

```diff
@@ -87,20 +87,41 @@ def sparse_coded_signal(rng):
     return X, y[:, 0], beta[:, 0]
 
 
-# TODO this fixture sometimes causes tests to fail because it does not pick some groups
 @pytest.fixture(params=[4, 6], scope="package")
 def random_model_with_groups(random_model, rng, request):
     """Add a correct set of groups to model."""
     X, y, beta = random_model
-    coef_mask = abs(beta) > 0
     n_groups = request.param
     n_active_groups = n_groups // 3 + 1
+
+    n_features_per_group = len(beta) // n_groups
     active_group_inds = rng.choice(range(n_groups), size=n_active_groups, replace=False)
     inactive_group_inds = np.setdiff1d(range(n_groups), active_group_inds)
 
     groups = np.zeros(len(beta), dtype=int)
-    for i, c in enumerate(coef_mask):
-        groups[i] = (
-            rng.choice(active_group_inds) if c else rng.choice(inactive_group_inds)
-        )
+    active_feature_inds = np.where(abs(beta) > 0)[0]
+    inactive_feature_inds = np.setdiff1d(np.arange(len(beta)), active_feature_inds)
```

Comment:
Did you add test to check whether the regularizations give correct results of active feature indices as well? If you did, could you please point out where I can find the tests?

Reply:
I have not implemented an "exact" results test. If you have any ideas on how to write a good and robust test I'd find that very helpful

Reply:
I have no idea yet. Maybe I'll do this in a follow up PR if I have an idea.